### PR TITLE
New website and documentation

### PR DESCRIPTION
**Pull request by lucperkins**
### Motivation

As it stands, all documentation for Pulsar resides in Markdown files in the `docs` directory. This PR provides an accompanying website, along with a CI setup for automatic publishing of the site.

### Modifications

The documentation content was initially derived from the Markdown material in `docs`, but has been greatly expanded.

### Result

Pulsar will have its own website and much-improved documentation. Although much work remains, I feel strongly that this is a solid first incarnation.

### CODE REVIEW

**Comment by merlimat** [comment]
@saandrews @joefk @rdhabalia Please take a look

**Comment by merlimat** [comment]
For review purposes, the generated website is temporarily available at https://streamlio.github.io/incubator-pulsar/

**Comment by yush1ga** [comment]
https://streamlio.github.io/incubator-pulsar/team/
In our brand guidelines, "Yahoo! Japan" is not allowed.
Could you replace it with "Yahoo Japan Corporation" ?

**Comment by merlimat** [comment]
@yush1ga Fixed the `team.md` page with correct name

**Comment by merlimat** [comment]
@joefk @saandrews @rdhabalia Please take a look ;) 

Once we get this one in, the build will generate the first version in the `asf-site` branch in the same repo. At that point we can ask INFRA to enable the gitpubsub bridge to publish the site. We can still do all sorts of changes fixes before (and after) that.

**Comment by saandrews** [comment]
Looks nice @merlimat :) Will go through and provide feedback.

**Comment by yush1ga** [comment]
@merlimat 👍

**Comment by maskit** [comment]
Can you remove my company name from the table? It's not a secret since it's also written on the project proposal (and I understand the necessity for it), but my contribution will not relate to my company at all.

**Comment by merlimat** [comment]
@maskit Fixed

**Comment by merlimat** [comment]
Any additional thoughts or concerns about this PR? I'd like to get the first cut out, so that can enabled the website, and later we can continue iterate on the content.

**Comment by rdhabalia** [comment]
should we add license to site/_sass/*scss files?

**Comment by rdhabalia** [comment]
README.md logo still points to `docs/img/pulsar.png` which is moved under `site`

File: site/presentations.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+title: Presentations
+layout: content
+---
```
Comment (rdhabalia) [inline site/presentations.md]: is this table expected in all `.md` file, may it's required in site generation?

File: site/index.html [hunk 2]
```diff
@@ -0,0 +1,77 @@
+---
+title: Apache Pulsar
+layout: default
```
Comment (rdhabalia) [inline site/index.html]: license doesn't present in beginning, will it fail in license check ?

File: site/docs/latest/index.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+layout: docs-main
+---
+
+Welcome to the documentation for [Pulsar]({{ site.baseurl }}), an open-source distributed pub-sub messaging system originally created by [Yahoo](http://yahoo.github.io) and now under the stewardship of the [Apache Software Foundation](http://www.apache.org/).
+
+## Translations
+
+The Pulsar documentation is available in the following languages:
+
+<img src="{{ site.baseurl }}img/japan-flag.svg" class="mt-0 mb-0" style="border: 1px solid #ddd; width: 4rem; margin-right: 1.25rem;"> [Japanese]({{ site.baseurl }}ja)
```
Comment (rdhabalia) [inline site/docs/latest/index.md]: are we missing correct img path here?

File: site/presentations.md (hunk 1 above)
Comment (merlimat) [inline site/presentations.md]: Yes, that's standard practice for all Markdown converters (in this case Jekyll) to have a YAML preamble with metadata

File: site/index.html (hunk 2 above)
Comment (merlimat) [inline site/index.html]: Yes, verified that still passes the check

File: site/docs/latest/index.md (hunk 3 above)
Comment (merlimat) [inline site/docs/latest/index.md]: I think it gets generated correctly, `site.baseUrl` gets expanded at generation time.

**Comment by merlimat** [comment]
> should we add license to site/_sass/*scss files?

I'm not sure that's 100% required, since they're not going to be included in the src or binary distribution, just exposed as website.

> README.md logo still points to docs/img/pulsar.png which is moved under site

Good point, this should be fixed. 
Also, after this commit, we should fix the README to include basic instructions on how to compile from sources.

**Comment by merlimat** [comment]
@rdhabalia Updated

**Pull request merged**